### PR TITLE
feat: [DHIS2-17589] show absolute date when hovering over relative dates

### PR DESCRIPTION
**Pull request by alaa-yahia**
Implement [DHIS2-17589](https://dhis2.atlassian.net/browse/DHIS2-17589)

[DHIS2-17589]: https://dhis2.atlassian.net/browse/DHIS2-17589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/components/CardList/CardListItem.component.js [hunk 1]
```diff
@@ -233,7 +233,10 @@ const CardListItemIndex = ({
                                 {
                                     item.tei && item.tei.lastUpdated &&
                                     <div className={classes.smallerLetters}>
-                                        { i18n.t('Last updated') } {item.tei && moment(item.tei.lastUpdated).fromNow()}
+                                        { i18n.t('Last updated') }
+                                        { item.tei && <Tooltip content={item.tei.lastUpdated.toLocaleString()}>
+                                            { moment(item.tei.lastUpdated).fromNow() }
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/CardList/CardListItem.component.js:238]: Let's add `fromServerDate` to properly handle the time difference between the client and the server.
When enrolling and the list of possible duplicates is displayed, `item.tei.lastUpdated` doesn't show me anything. Does it work for you? I could only see the date when I used `item.tei.updatedAt` instead.

File: src/core_modules/capture-core/components/WidgetEnrollment/WidgetEnrollment.component.js [hunk 2]
```diff
@@ -148,9 +149,10 @@ export const WidgetEnrollmentPlain = ({
                             <span className={classes.icon} data-test="widget-enrollment-icon-clock">
                                 <IconClock16 color={colors.grey600} />
                             </span>
-                            {i18n.t('Last updated {{date}}', {
-                                date: moment(fromServerDate(enrollment.updatedAt)).fromNow(),
-                            })}
+                            {i18n.t('Last updated')}
+                            <Tooltip content={(fromServerDate(enrollment.updatedAt).toLocaleString())}>
+                                {moment(fromServerDate(enrollment.updatedAt)).fromNow()}
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetEnrollment/WidgetEnrollment.component.js:154]: After I edit the enrollment, I see the text `Last updated in 2 hours` and the tooltip shows a date that is 2 hours away. The bug is happening in play too. Can you take a look to see if this time zone discrepancy is easy to fix in this PR or if we need to open another bug ticket?  Thanks!
Screenshot 2024-07-02 at 07 42 09

**Review by simonadomnisoru — COMMENTED** [review]
Hi,
Can you also add the tooltip in the [events and enrollment notes](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetComment/CommentSection/CommentSection.js#L107)? Something like this:
```
<Tooltip content={fromServerDate(storedAt).toLocaleString()}>
    {moment(fromServerDate(storedAt)).fromNow()}
</Tooltip>
```
Thanks!

File: src/core_modules/capture-core/components/CardList/CardListItem.component.js (hunk 1 above)
Comment (alaa-yahia) [reply src/core_modules/capture-core/components/CardList/CardListItem.component.js:238]: No, it doesn't show unless `updatedAt` is used. Will update the code

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Comment by linadhis2** [comment]
LGTM from QA perspective

**Pull request merged**